src/lexer/mod.rs

```diff
@@ -106,6 +106,7 @@ impl<'cache, 'contents> Lexer<'cache, 'contents> {
             ("float", Token::FloatType),
             ("char", Token::CharType),
             ("string", Token::StringType),
+            ("ptr", Token::PointerType),
```

Comment:
Nitpick: Can we change the keyword to a capital `Ptr` instead? I've been debating whether all types should be capitalized but am still undecided. Since `Ptr` being a keyword may be temporary lets make the keyword capitalized to match its eventual library definition.

Reply:
It seemed to try to resolve to a user defined type if it was a capital P. I can try to see if i can change the parsing order for it to resolve correctly though :)